src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,263 @@
+          @click="${this._handleModalClose}"
+          @keydown="${this._handleOverlayKeyup}"
+        >
+          <div
+            id="container"
+            role="dialog"
+            aria-modal="true"
+            aria-labelledby="${ifDefined(this._getHeaderTitleSlotId())}"
+            aria-describedby="${ifDefined(
+              this._getAccessibilityDescriptionId()
+            )}"
+          >
+            <div id="header">
+              <slot id="title" name="outline-modal--header"></slot>
+              <button
+                id="close"
+                aria-label="Close modal"
+                @click="${this._handleModalClose}"
+                @keydown="${this._handleModalCloseKeyup}"
+              ></button>
+            </div>
+            <div id="main">
+              <slot></slot>
+            </div>
+          </div>
+        </div>
+        <slot
+          id="accessibility-description"
+          name="outline-modal--accessibility-description"
+        ></slot>
+      `;
+    }
+
+    return template;
+  }
+
+  private _getHeaderTitleSlotId(): string | null {
+    let id = null;
+
+    // When this was a class property, it wasn't finding the slot as expected.
+    const headerSlot: HTMLSlotElement | null = this.querySelector(
+      '[slot="outline-modal--header"]'
+    );
+
+    if (headerSlot !== null) {
+      id = 'header';
+    }
+
+    return id;
+  }
+
+  private _getAccessibilityDescriptionId(): string | null {
+    let id = null;
+
+    // When this was a class property, it wasn't finding the slot as expected.
+    const accessibilityDescriptionSlot: HTMLSlotElement | null =
+      this.querySelector('[slot="outline-modal--accessibility-description"]');
+
+    if (accessibilityDescriptionSlot !== null) {
+      id = 'accessibility-description';
+    }
+
+    return id;
+  }
+
+  async open(): Promise<void> {
+    if (!this.isOpen) {
+      this.isOpen = true;
+
+      await this.updateComplete;
+
+      this._focusOnModalElement();
+
+      this._trapFocusWithinModal();
+
+      this.dispatchEvent(new CustomEvent('opened'));
+    }
+  }
+
+  async close(): Promise<void> {
+    if (this.isOpen) {
+      this.isOpen = false;
+
+      await this.updateComplete;
+
+      this.dispatchEvent(new CustomEvent('closed'));
+
+      this.triggerElement.focus();
+    }
+  }
+
+  @query('#trigger')
+  private triggerElement!: HTMLDivElement;
+
+  private _handleModalTrigger(event: MouseEvent | KeyboardEvent): void {
+    let shouldOpen = false;
+
+    switch (event.type) {
+      case 'click':
+        shouldOpen = true;
+        break;
+      case 'keydown':
+        if ('key' in event && event.key === 'Enter') {
+          shouldOpen = true;
+          // This prevents a focused element from also triggering.
+          // For example, the modal opens and the "accept" button is focused and then triggered and the modal closes.
+          event.preventDefault();
+          break;
+        }
+    }
```

Comment:
I think I should probably split this into two handlers. One for the MouseEvent and one for the KeyboardEvent.

The code is a bit difficult to read as-is, because it takes two different inputs, spends most of the time figuring out which of those we are looking at, and then does something very simple.

I'll try to give this a shot on Monday.